solutions/usaco-739.mdx

```diff
@@ -0,0 +1,142 @@
+---
+id: usaco-739
+source: USACO Silver 2017 US Open 
+title: Bovine Genomics
+author: Brad Ma
+---
+
+[Official Editorial (C++)](http://www.usaco.org/current/data/sol_cownomics_silver_open17.html)
+
+## Implementation
+
+**Time Complexity** $\mathcal{O}(N^3)$
+
+<LanguageSection>
+  
+<JavaSection>
+  
+<Warning>
+This implementation uses the set data structure, which has not been covered yet. See [Introduction to Sets/Maps](https://usaco.guide/bronze/intro-sets) for more details about sets.
+</Warning>
+  
+```java
+import java.io.*;
+import java.util.*;
+
+public class BovineGenomics {
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio();
+		//BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		//PrintWriter pw = new PrintWriter(System.out);
+
+		int n = io.nextInt();
+		int m = io.nextInt();
+
+		int[][] spotted = new int[n][m];
+		for (int x = 0; x < n; x++) {  // read spotted
+			StringBuilder line = new StringBuilder();
+			line.append(io.next());
+
+			for (int y = 0; y < m; y++) {
+				// using base 4 will assign a unique
+				// number to every string
+				// A -> 1, C -> 4, T -> 16, G -> 64
+				if (line.charAt(y) == 'A') {
+					spotted[x][y] = 0;
+				}
+				else if (line.charAt(y) == 'C') {
+					spotted[x][y] = 1;
+				}
+				else if (line.charAt(y) == 'T') {
+					spotted[x][y] = 2;
+				}
+				else {
+					spotted[x][y] = 3;
+				}
+			}
+		}
+
+		int[][] plain = new int[n][m];
+		for (int x = 0; x < n; x++) {  // read plain
+			StringBuilder line = new StringBuilder();
+			line.append(io.next());
+
+			for (int y = 0; y < m; y++) {
+				// A -> 1, C -> 4, T -> 16, G -> 64
```

Comment:
maybe explain what these numbers mean

Reply:
also, it doesn't seem like you're using the numbers here (if you are, it's not obvious to the reader)

Reply:
will multiply later on, added this as a comment